x-pack/plugins/security_solution/public/overview/containers/overview_cti_links/helpers.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObjectAttributes } from '@kbn/securitysolution-io-ts-alerting-types';
+import { CTI_DEFAULT_SOURCES } from '../../../../common/cti/constants';
+
+export interface CtiListItem {
+  path: string;
+  title: string;
+  count: number;
+}
+
+export const EMPTY_LIST_ITEMS: CtiListItem[] = CTI_DEFAULT_SOURCES.map((item) => ({
+  title: item,
+  count: 0,
+  path: '',
+}));
+
+export const TAG_REQUEST_BODY = {
+  type: 'tag',
+  search: 'threat intel',
+  searchFields: ['name'],
+};
+
+export interface EventCounts {
+  [key: string]: number;
+}
+
+export const DASHBOARD_SO_TITLE_PREFIX = '[Filebeat Threat Intel] ';
+export const OVERVIEW_DASHBOARD_LINK_TITLE = 'Overview';
+
+export const getListItemsWithoutLinks = (eventCounts: EventCounts): CtiListItem[] => {
+  return EMPTY_LIST_ITEMS.map((item) => ({
+    ...item,
+    count: eventCounts[item.title.replace(' ', '').toLowerCase()] ?? 0,
+  }));
+};
+
+export const isCtiListItem = (item: CtiListItem | Partial<CtiListItem>): item is CtiListItem =>
+  typeof item.title === 'string' && typeof item.path === 'string' && typeof item.count === 'number';
+
+export const isOverviewItem = (item: { path?: string; title?: string }) =>
+  item.title === OVERVIEW_DASHBOARD_LINK_TITLE;
+
+export const createLinkFromDashboardSO = (
```

Comment:
This logic seems brittle to dashboard changes. Have you thought about ways to make this integration more robust? Do we have tests that will fail if e.g. the dashboard titles are changed?

Reply:
great question. I got confirmation from Andrew Pease regarding the relationship between dashboard names and the corresponding `threatintel.` field names, that we can consistently assume that removal of spaces and lowercasing for the title will correspond to the field name. I definitely agree that this is a suboptimal approach, but as of this moment there is no other information on the dashboard object that enables us to infer which SO corresponds to which source. we should update this logic in the close feature, created elastic/kibana#103365 to keep track of the change